pkgs/development/interpreters/ruby/default.nix

```diff
@@ -187,6 +187,11 @@ let
               sed -i '/CC_VERSION_MESSAGE/d' $rbConfig
             ''
           }
+
+          # Allow to override compiler. This is important for cross compiling as
+          # we need to set a compiler that is different from the build one.
+          sed -i 's/CONFIG\["CC"\] = "\(.*\)"/CONFIG["CC"] = ENV["CC"] || "\1"/'  "$rbConfig"
```

Comment:
```suggestion
  awk -i inplace -F' = ' \
    ' # operate on the line starting with
      /^  CONFIG\["CC"\]/ {
        # replace the right hand side
        sub($2, "ENV[\"CC\"] || \"1\"")
      }; { print }' "$rbConfig"
  # test that the line isn't mangled in case upstream made the above unnecessary
  grep -qx '  CONFIG\["CC"\] = ENV\["CC"\] || "1"' "$rbConfig"
```